src/login/logind-dbus.c

```diff
@@ -2038,6 +2042,68 @@ static usec_t nologin_timeout_usec(usec_t elapse) {
         return LESS_BY(elapse, 5 * USEC_PER_MINUTE);
 }
 
+int manager_load_scheduled_shutdown(Manager *m) {
+        _cleanup_fclose_ FILE *f = NULL;
+        int r;
+
+        f = fopen(SHUTDOWN_SCHEDULE_FILE, "r");
+        if (!f) {
+                if (errno == ENOENT)
+                        return 0;
+                return -errno;
+        }
+
+        for (;;) {
+                _cleanup_free_ char *line = NULL;
+                const char *val, *l;
+
+                r = read_line(f, LONG_LINE_MAX, &line);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to read serialization line: %m");
+                if (r == 0)
+                        break;
+
+                l = strstrip(line);
+                if (isempty(l)) /* end marker */
+                        break;
+
+                if ((val = startswith(l, "USEC="))) {
+                        if (deserialize_usec(val, &m->scheduled_shutdown_timeout) < 0)
+                                log_notice("Failed to parse scheduled shutdown time '%s', ignoring.", val);
+                } else if ((val = startswith(l, "WARN_WALL="))) {
+                        if (safe_atou(val, &m->enable_wall_messages) < 0)
+                                log_notice("Failed to parse enable wall messages '%s', ignoring.", val);
+                } else if ((val = startswith(l, "MODE="))) {
+                        HandleAction handle = handle_action_from_string(val);
+                        if (handle == _HANDLE_ACTION_INVALID)
+                                log_notice("Failed to parse scheduled shutdown type '%s', ignoring.", val);
+                        else
+                                m->scheduled_shutdown_type = manager_item_for_handle(handle);
+                } else if ((val = startswith(l, "WALL_MESSAGE="))) {
+                        _cleanup_free_ char *unescaped = NULL;
+                        r = cunescape(val, 0, &unescaped);
+                        if (r >= 0)
+                                r = free_and_strdup(&m->wall_message, empty_to_null(unescaped));
+                        if (r < 0)
+                                return log_oom();
```

Comment:
style suggestion:
```
if (r < 0)
        return log_oom();

r = free_and_strdup(&m->wall_message, empty_to_null(unescaped));
```

Reply:
`free_and_strdup_warn()` can be used.

Reply:
Would like this then:
```
r = cunescape(val, 0, &unescaped);                                                    
if (r < 0)                                                                            
        return log_oom();                                                             
                                                                                      
r = free_and_strdup_warn(&m->wall_message, empty_to_null(unescaped));                 
if (r < 0)                                                                            
        return r;
```

We miss cunescape to static buffer :)